web/modules/weather_data/src/Service/AlertUtility.php

```diff
@@ -638,4 +638,34 @@ public static function getPlacesFromAlertDescription($alert)
         // so return false so we don't do any other processing elsewhere.
         return false;
     }
+
+    /* Determine the "level" of the alert (warning, watch, advisory, etc..)
+     * based on the the last word in the alert event name
+     * */
+    public static function getAlertLevel($event)
+    {
+        $eventWords = explode(" ", $event);
+        $alertLevel = strtolower($eventWords[array_key_last($eventWords)]);
+        return $alertLevel;
+    }
+
+    /* Determine the highest alert level from an array of alerts.
+     * Warnings are highest, followed by Watches, and anything else is last ("other").
+     * */
+    public static function getHighestAlertLevel($alerts)
+    {
+        if (count($alerts) == 0) {
+            return "";
+        }
+        $highestAlertLevel = "other";
+        foreach ($alerts as $alert) {
+            if ($alert->alertLevel == "warning") {
+                $highestAlertLevel = "warning";
+                break;
```

Comment:
Nice! I have deep-seated feelings about flow-control statements like `break`, but this seems like a great way to use it.